arcgis-ios-sdk-samples/Layers/Display OGC API collection/DisplayOGCAPICollectionViewController.swift

```diff
@@ -0,0 +1,107 @@
+        // Set a limit of 5000 on the number of returned features per request,
+        // because the default on some services could be as low as 10.
+        queryParameters.maxFeatures = 5_000
+        return queryParameters
+    }()
+    
+    /// A feature layer to visualize the OGC API features.
+    let ogcFeatureLayer: AGSFeatureLayer = {
+        // Note: the collection ID can be accessed later via
+        // `featureCollectionInfo.collectionID` property of the feature table.
+        let table = AGSOGCFeatureCollectionTable(
+            url: URL(string: "https://demo.ldproxy.net/daraa")!,
+            collectionID: "TransportationGroundCrv"
+        )
+        // Set the feature request mode to manual (only manual is currently
+        // supported). In this mode, you must manually populate the table -
+        // panning and zooming won't request features automatically.
+        table.featureRequestMode = .manualCache
+        
+        let featureLayer = AGSFeatureLayer(featureTable: table)
+        let lineSymbol = AGSSimpleLineSymbol(style: .solid, color: .blue, width: 3)
+        featureLayer.renderer = AGSSimpleRenderer(symbol: lineSymbol)
+        return featureLayer
+    }()
+    
+    // MARK: Method
+    
+    func populateFeaturesFromQuery() {
+        // Cancel if there is an existing query request.
+        lastQuery?.cancel()
+        
+        // Set the query's geometry to the current visible extent.
+        visibleExtentQueryParameters.geometry = mapView.visibleArea?.extent
+        
+        // Populate the table with the query. Setting `outFields` to `nil`
+        // requests all fields.
+        let table = ogcFeatureLayer.featureTable as! AGSOGCFeatureCollectionTable
+        lastQuery = table.populateFromService(
+            with: visibleExtentQueryParameters,
+            clearCache: false,
+            outfields: nil
+        ) { [weak self] _, error in
+            if let error = error,
+               // Do not display error if user simply cancelled the request.
+               (error as NSError).code != NSUserCancelledError {
+                self?.presentAlert(error: error)
+            }
+        }
+    }
+    
+    // MARK: UIViewController
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["DisplayOGCAPICollectionViewController"]
+        // Query for features with the initial extent.
+        populateFeaturesFromQuery()
+        // Query for features whenever map's viewpoint change has finished.
+        navigatingObservation = mapView.observe(\.isNavigating, options: .new) { _, change in
+            if change.newValue == false {
+                DispatchQueue.main.async {
+                    self.populateFeaturesFromQuery()
+                }
+            }
+        }
+    }
```

Comment:
A few things:

- Boolean values are, well, Boolean values. They should not be tested for equality using `== [true/false]`.
- The first call to `populateFeaturesFromQuery()` can be avoided (or rather, made implicit) by passing `initial` as one of the options.
- The change dictionary isn't needed as the new value can be read directly from the map view, as supplied by the closure. Doing so prevents the change dictionary from being created, which is a good thing if it isn't necessary.
- `self` could potentially go away before `populateFeaturesFromQuery()`, so the closure should weakly capture `self` so as not to keep it around longer than necessary.
- 
```suggestion
        // Query for features whenever the user stops interacting with the map
        // view.
        navigatingObservation = mapView.observe(\.isNavigating, options: .initial) { mapView, _ in
            guard !mapView.isNavigating else { return }
            DispatchQueue.main.async { [weak self] in
                self?.populateFeaturesFromQuery()
            }
        }
```

Reply:
Great suggestions. 👍